File: Applications/Escrow.BOAS.Web/Scripts/Areas/BrokerManagement/Controllers/BankInformationController.js
//CREATED BY: Rifad
//Date: 6-5-2017
//
//
angular.module('myApp').controller('BankInformationController', function ($scope, $window, $http, $location, $routeParams, $filter, Upload, dateconfigservice, globalvalueservice, isundefinedornullservice) {

    $('#pageTitle').text("--- Bank Information");

    $scope.image = '../Images/images.png';

    $scope.open = function ($event, opened) {


        $event.preventDefault();
        $event.stopPropagation();

        $scope[opened] = true;

    };

    $scope.loadBankList = function () {
        $scope.rowCollection = [];
        $http.get('/BrokerManagement/BankInformation/getBankList/').
          success(function (data) {
              if (data.success) {
                  $scope.rowCollection = data.data;
                  $scope.displayedCollection = [].concat($scope.rowCollection);
              }
              else {
                  toastr.error(data.errorMessage);
              }
          }).
          error(function (XMLHttpRequest, textStatus, errorThrown) {
              toastr.error(XMLHttpRequest + ": " + textStatus + ": " + errorThrown, 'Error!!!');
          });
    };

    $scope.refresh = function () {
        $scope.StatusList = "";
        $scope.bankInformation.short_name = "";
        $scope.bankInformation.bank_name = "";
        //  $scope.bankInformation.active_status_id = "";
        $scope.LoadAllDropDown();

    };
    $scope.saveNewBank = function () {

        if ($scope.newBankForm.$valid) {
            $http.post('/BrokerManagement/BankInformation/AddNewBank/', $scope.bankInformation
            ).
                 success(function (data) {
                     if (data.success) {
                         toastr.success("Submitted Successfully");
                         $scope.rowCollection = data.data;
                     }
                     else {
                         toastr.error(data.errorMessage);
                     }
                 }).
           error(function (XMLHttpRequest, textStatus, errorThrown) {
               toastr.error(XMLHttpRequest + ": " + textStatus + ": " + errorThrown, 'Error!!!');
           });

        }
    };

    $scope.LoadAllDropDown = function () {
        $scope.ActiveStatus();

        $scope.DistrictName();
    };
    $scope.ActiveStatus = function () {
        $scope.StatusList = null;
        $http.get('/BrokerManagement/BankInformation/GetStatus/')
        .success(function (data) {
            $scope.StatusList = data;
        }).
      error(function (XMLHttpRequest, textStatus, errorThrown) {
          toastr.error(XMLHttpRequest + ": " + textStatus + ": " + errorThrown, 'Error!!!');
      });
    }

    $scope.loadBankById = function () {
        $scope.bankInformation = {};
        $http.get('/BrokerManagement/BankInformation/GetBankById/' + $routeParams.id).
          success(function (data) {
              if (data.success) {
                  $scope.bankInformation = data.data;
               //   $scope.LoadAllDropDown();
              }
              else {
                  toastr.error(data.errorMessage);
              }
          }).
          error(function (XMLHttpRequest, textStatus, errorThrown) {
              toastr.error(XMLHttpRequest + ": " + textStatus + ": " + errorThrown, 'Error!!!');
          });
    };


    $scope.updateBank = function (id) {
        if ($scope.newBankForm.$valid) {
            $scope.bankInformation.id = id;
            $http.post('/BrokerManagement/BankInformation/UpdateBank/', $scope.bankInformation
            ).
                 success(function (data) {
                     if (data.success) {
                         toastr.success("Update Successfully");
                     }
                     else {
                         toastr.error(data.errorMessage);
                     }
                 }).
           error(function (XMLHttpRequest, textStatus, errorThrown) {
               toastr.error(XMLHttpRequest + ": " + textStatus + ": " + errorThrown, 'Error!!!');
           });

        }
    };

    $scope.DeleteBank = function (index) {

        // $id = 0;
        $scope.bankInformation = $scope.displayedCollection[index];

        $http.post('/BrokerManagement/BankInformation/DeleteBank/', $scope.bankInformation
           ).
                success(function (data) {
                    if (data.success) {
                        toastr.success("Delete Successfully");
                        //$scope.displayedCollection.splice(index, 1);
                        $scope.loadBankList();
                    }
                    else {
                        toastr.error(data.errorMessage);
                    }
                }).
          error(function (XMLHttpRequest, textStatus, errorThrown) {
              toastr.error(XMLHttpRequest + ": " + textStatus + ": " + errorThrown, 'Error!!!');
          });
    };
    $scope.loadBranchList = function () {
        $scope.bank_id = $routeParams.id;
        $scope.bankInformation = {};
        $http.get('/BrokerManagement/BankInformation/GetBranchByBankId/' + $routeParams.id).
          success(function (data) {
              if (data.success) {
                  $scope.rowCollection = data.data;
                  $scope.displayedCollection = [].concat($scope.rowCollection);
                  $scope.loadBankById();
              }
              else {
                  toastr.error(data.errorMessage);
              }
          }).
          error(function (XMLHttpRequest, textStatus, errorThrown) {
              toastr.error(XMLHttpRequest + ": " + textStatus + ": " + errorThrown, 'Error!!!');
          });
    };
    $scope.loadBranchById = function () {
        $http.get('/BrokerManagement/BankInformation/GetBranchById/' + $routeParams.id)
          .success(function (data) {
              if (data.success) {
                  $scope.branchInformation = data.data;
              }
              else {
                  toastr.error(data.errorMessage);
              }
          }).
          error(function (XMLHttpRequest, textStatus, errorThrown) {
              toastr.error(XMLHttpRequest + ": " + textStatus + ": " + errorThrown, 'Error!!!');
          });
    };

    $scope.DistrictName = function () {
        $scope.StatusList = null;
        $http.get('/BrokerManagement/BankInformation/DistrictName/')
        .success(function (data) {
            $scope.DistrictList = data;
        }).
      error(function (XMLHttpRequest, textStatus, errorThrown) {
          toastr.error(XMLHttpRequest + ": " + textStatus + ": " + errorThrown, 'Error!!!');
      })

    };
    $scope.updateBranch = function (id) {
        if ($scope.newBranchForm.$valid) {
            $scope.branchInformation.id = id;
            $http.post('/BrokerManagement/BankInformation/UpdateBranch/', $scope.branchInformation
            ).
                 success(function (data) {
                     if (data.success) {
                         toastr.success("Update Successfully");
                     }
                     else {
                         toastr.error(data.errorMessage);
                     }
                 }).
           error(function (XMLHttpRequest, textStatus, errorThrown) {
               toastr.error(XMLHttpRequest + ": " + textStatus + ": " + errorThrown, 'Error!!!');
           });

        }
    }
    $scope.DeleteBranch = function (index) {
        $scope.branchInformation = $scope.displayedCollection[index];

        $http.post('/BrokerManagement/BankInformation/DeleteBranch/', $scope.branchInformation
           ).
                success(function (data) {
                    if (data.success) {
                        toastr.success("Delete Successfully");
                        //$scope.displayedCollection.splice(index, 1);
                        $scope.loadBranchList();
                    }
                    else {
                        toastr.error(data.errorMessage);
                    }
                }).
          error(function (XMLHttpRequest, textStatus, errorThrown) {
              toastr.error(XMLHttpRequest + ": " + textStatus + ": " + errorThrown, 'Error!!!');
          });
    }


    $scope.KeepBankId = function () {
        $scope.branchInformation = {};
        $scope.branchInformation.bank_id = $routeParams.id;
        $scope.LoadAllDropDown();
    };


    $scope.saveNewBranch = function () {

        if ($scope.newBranchForm.$valid) {
            $http.post('/BrokerManagement/BankInformation/AddNewBranch/', $scope.branchInformation
            ).
                 success(function (data) {
                     if (data.success) {
                         toastr.success("Submitted Successfully");
                     }
                     else {
                         toastr.error(data.errorMessage);
                     }
                 }).
           error(function (XMLHttpRequest, textStatus, errorThrown) {
               toastr.error(XMLHttpRequest + ": " + textStatus + ": " + errorThrown, 'Error!!!');
           });

        }
    };
    $scope.refreshBranch = function () {
        branchInformation.branch_name = "";
        branchInformation.address = "";
        branchInformation.email_address = "";
        branchInformation.contact_person = "";
        branchInformation.contact_number = "";
        branchInformation.routing_no = "";
    }


});

File: Applications/Escrow.BOAS.Web/Areas/BrokerManagement/Controllers/BankInformationController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using Microsoft.AspNet.Identity;
using Microsoft.AspNet.Identity.EntityFramework;
using Microsoft.Owin.Security;
using Escrow.BOAS.Web.Models;
using Escrow.BOAS.BrokerManagement.Models;
using Escrow.BOAS.BrokerManagement.Factories;
using Escrow.BOAS.Utility;
using Microsoft.AspNet.Identity.Owin;
using Microsoft.Owin.Security.DataProtection;
using Escrow.Utility.Interfaces;
using System.Data.Entity;
using Escrow.BOAS.Configuration.Factories;
///=============================================================
///Created by: Rifad
///Created date: 6/5/2017
///Reason: Bank list, add, edit, delete, Branch list add new, Update, Delete
///=============================================================
namespace Escrow.BOAS.Web.Areas.BrokerManagement.Controllers
{
    public class BankInformationController : Controller
    {

        // GET: BrokerManagement/BankInformation
        private static IGenericFactory<Escrow.BOAS.AccountingManagement.Models.tblBank> bankFactory;
        private static IGenericFactory<Escrow.BOAS.AccountingManagement.Models.tblBankBranch> branchFactory;

        // private IGenericFactory
        public ActionResult BankList()
        {
            return View();
        }
        public JsonResult getBankList()
        {
            try
            {
                decimal membership_id = SessionManger.BrokerOfLoggedInUser(Session).membership_id;
                bankFactory = new Escrow.BOAS.AccountingManagement.Factories.BankFactory();

                var bankList = bankFactory.GetAll()
                    .Select(s => new
                    {
                        id = s.id,
                        short_name = s.short_name,
                        bank_name = s.bank_name
                    }).ToList();

                return Json(new { success = true, data = bankList }, JsonRequestBehavior.AllowGet);
            }
            catch (Exception ex)
            {
                return Json(new { success = false, errorMessage = ex.Message }, JsonRequestBehavior.AllowGet);
            }
        }

        public ActionResult NewBankInformation()
        {
            return View();
        }
        public JsonResult AddNewBank(Escrow.BOAS.AccountingManagement.Models.tblBank bankInformation)
        {
            try
            {
                decimal membership_id = SessionManger.BrokerOfLoggedInUser(Session).membership_id;
                bankFactory = new Escrow.BOAS.AccountingManagement.Factories.BankFactory();
                Escrow.BOAS.AccountingManagement.Models.tblBank bankTableObj = new Escrow.BOAS.AccountingManagement.Models.tblBank();
                bankTableObj = bankFactory.FindBy(a => a.short_name == bankInformation.short_name && a.bank_name == bankInformation.bank_name).FirstOrDefault();

                if (bankTableObj == null)
                {

                    bankInformation.membership_id = membership_id;
                    bankInformation.changed_user_id = User.Identity.GetUserId();
                    bankInformation.changed_date = DateTime.Now;
                    bankFactory.Add(bankInformation);
                    bankFactory.Save();

                    return Json(new { success = true, data = "Successfully Submited" }, JsonRequestBehavior.AllowGet);
                }
                else
                {
                    return Json(new { success = false, errorMessage = "Bank already exists!!!!" }, JsonRequestBehavior.AllowGet);
                }
            }
            catch (Exception ex)
            {
                return Json(new { success = false, errorMessage = ex.Message }, JsonRequestBehavior.AllowGet);
            }
        }
        public JsonResult GetStatus()
        {
            var data = DropDown.ddlActiveStatus();
            return Json(data, JsonRequestBehavior.AllowGet);
        }
        public ActionResult editBankInfomation(Escrow.BOAS.AccountingManagement.Models.tblBank bankInformation)
        {
            return View();
        }
        public JsonResult GetBankById(int id)
        {
            try
            {
                decimal membership_id = SessionManger.BrokerOfLoggedInUser(Session).membership_id;
                bankFactory = new Escrow.BOAS.AccountingManagement.Factories.BankFactory();
                var bankList = bankFactory.GetAll().AsEnumerable().Where(x => x.id == id)
                    .Select(s => new
                    {
                        id = s.id,
                        short_name = s.short_name,
                        bank_name = s.bank_name,
                        active_status_id = s.active_status_id
                    }).FirstOrDefault();

                return Json(new { success = true, data = bankList }, JsonRequestBehavior.AllowGet);
            }
            catch (Exception ex)
            {
                return Json(new { success = false, errorMessage = ex.Message }, JsonRequestBehavior.AllowGet);
            }
        }
        public JsonResult UpdateBank(Escrow.BOAS.AccountingManagement.Models.tblBank bankInformation)
        {
            try
            {
                decimal membership_id = SessionManger.BrokerOfLoggedInUser(Session).membership_id;
                bankFactory = new Escrow.BOAS.AccountingManagement.Factories.BankFactory();
                Escrow.BOAS.AccountingManagement.Models.tblBank bankTableObj = new Escrow.BOAS.AccountingManagement.Models.tblBank();
                bankTableObj = bankFactory.FindBy(a => a.short_name == bankInformation.short_name && a.bank_name == bankInformation.bank_name).FirstOrDefault();
                //bankTableObj = bankFactory.FindBy(a => a.short_name == bankInformation.short_name && a.bank_name == bankInformation.bank_name).FirstOrDefault();

                if (bankTableObj == null)
                {

                    bankInformation.membership_id = membership_id;
                    bankInformation.changed_user_id = User.Identity.GetUserId();
                    bankInformation.changed_date = DateTime.Now;
                    bankFactory.Edit(bankInformation);
                    bankFactory.Save();
                    return Json(new { success = true, data = "Update Successfully" }, JsonRequestBehavior.AllowGet);
                }
                else
                {
                    return Json(new { success = false, errorMessage = "Bank already exists!!!!" }, JsonRequestBehavior.AllowGet);
                }

            }
            catch (Exception ex)
            {
                return Json(new { success = false, errorMessage = ex.Message }, JsonRequestBehavior.AllowGet);
            }
        }

        public JsonResult DeleteBank(Escrow.BOAS.AccountingManagement.Models.tblBank bankInformation)
        {
            try
            {
                bankFactory = new Escrow.BOAS.AccountingManagement.Factories.BankFactory();
                var bank = bankFactory.GetById(bankInformation.id);
                bankFactory.Delete(bank);
                bankFactory.Save();
                return Json(new { message = "Successfully Deleted", success = true }, JsonRequestBehavior.AllowGet);
            }
            catch (Exception ex)
            {
                return Json(new { success = false, errorMessage = ex.Message }, JsonRequestBehavior.AllowGet);
            }
        }
        public ActionResult BranchList()
        {
            return View();
        }
        public JsonResult GetBranchByBankId(int id)
        {
            try
            {
                branchFactory = new Escrow.BOAS.AccountingManagement.Factories.BankBranchFactory();
                var branchList = branchFactory.GetAll().AsEnumerable().Where(a => a.bank_id == id)
                    .Select(b => new
                    {
                        id = b.id,
                        bank_id = b.bank_id,
                        branch_name = b.branch_name,
                        address = b.address,
                        email_address = b.email_address,
                        contact_person = b.contact_person,
                        contact_number = b.contact_number,
                        district_name = b.tblConstantObjectValue.display_value,
                        routing_no = b.routing_no
                    }).ToList();
                return Json(new { success = true, data = branchList }, JsonRequestBehavior.AllowGet);
            }
            catch (Exception ex)
            {
                return Json(new { success = false, errorMessage = ex.Message }, JsonRequestBehavior.AllowGet);
            }
        }
        public JsonResult DistrictName()
        {
            var data = DropDown.ddlDistrict();

            return Json(data, JsonRequestBehavior.AllowGet);
        }

          public ActionResult editBranchInfomation(Escrow.BOAS.AccountingManagement.Models.tblBankBranch branchInformation)
        {
            return View();
        }
          public JsonResult GetBranchById(int id)
          {
              try
              {
                  decimal membership_id = SessionManger.BrokerOfLoggedInUser(Session).membership_id;
                  branchFactory = new Escrow.BOAS.AccountingManagement.Factories.BankBranchFactory();
                  var branchList = branchFactory.GetAll().AsEnumerable().Where(x => x.id == id)
                      .Select(b => new
                      {
                          id = b.id,
                          bank_id = b.bank_id,
                          branch_name = b.branch_name,
                          address = b.address,
                          email_address = b.email_address,
                          contact_person = b.contact_person,
                          contact_number = b.contact_number,
                          routing_no = b.routing_no,
                          active_status_id = b.active_status_id,
                          district_id = b.district_id
                      }).FirstOrDefault();

                  return Json(new { success = true, data = branchList }, JsonRequestBehavior.AllowGet);
              }
              catch (Exception ex)
              {
                  return Json(new { success = false, errorMessage = ex.Message }, JsonRequestBehavior.AllowGet);
              }
          }
          public JsonResult UpdateBranch(Escrow.BOAS.AccountingManagement.Models.tblBankBranch branchInformation)
          {
              try
              {
                  decimal membership_id = SessionManger.BrokerOfLoggedInUser(Session).membership_id;
                  branchFactory = new Escrow.BOAS.AccountingManagement.Factories.BankBranchFactory();
                  Escrow.BOAS.AccountingManagement.Models.tblBankBranch branchTableObj = new Escrow.BOAS.AccountingManagement.Models.tblBankBranch();
                  branchTableObj = branchFactory.FindBy(a => a.branch_name == branchInformation.branch_name && a.routing_no == branchInformation.routing_no && a.district_id == branchInformation.district_id).FirstOrDefault();
                  if (branchTableObj == null)
                  {
                      branchInformation.membership_id = membership_id;
                      branchInformation.changed_user_id = User.Identity.GetUserId();
                      branchInformation.changed_date = DateTime.Now;
                      branchFactory.Edit(branchInformation);
                      branchFactory.Save();
                      return Json(new { success = true, data = "Update Successfully" }, JsonRequestBehavior.AllowGet);
                  }
                  else
                  {
                      return Json(new { success = false, errorMessage = "Branch already exists!!!!" }, JsonRequestBehavior.AllowGet);
                  }

              }
              catch (Exception ex)
              {
                  return Json(new { success = false, errorMessage = ex.Message }, JsonRequestBehavior.AllowGet);
              }
          }
          public JsonResult DeleteBranch(Escrow.BOAS.AccountingManagement.Models.tblBankBranch branchInformation)
          {
              try
              {
                  branchFactory = new Escrow.BOAS.AccountingManagement.Factories.BankBranchFactory();
                  var branch = branchFactory.GetById(branchInformation.id);
                  branchFactory.Delete(branch);
                  branchFactory.Save();
                  return Json(new { message = "Successfully Deleted", success = true }, JsonRequestBehavior.AllowGet);
              }
              catch (Exception ex)
              {
                  return Json(new { success = false, errorMessage = ex.Message }, JsonRequestBehavior.AllowGet);
              }
          }
          public ActionResult AddBranch()
          {
              return View();
          }
          public JsonResult AddNewBranch(Escrow.BOAS.AccountingManagement.Models.tblBankBranch branchInformation)
          {
              try
              {
                  decimal membership_id = SessionManger.BrokerOfLoggedInUser(Session).membership_id;
                  branchFactory = new Escrow.BOAS.AccountingManagement.Factories.BankBranchFactory();
                  Escrow.BOAS.AccountingManagement.Models.tblBankBranch branchTableObj = new Escrow.BOAS.AccountingManagement.Models.tblBankBranch();
                  branchTableObj = branchFactory.FindBy(a =>a.email_address==branchInformation.email_address&& a.address==branchInformation.address&&a.routing_no==branchInformation.routing_no && a.branch_name == branchInformation.branch_name && a.routing_no == branchInformation.routing_no && a.district_id == branchInformation.district_id).FirstOrDefault();

                  if (branchTableObj == null)
                  {

                      branchInformation.membership_id = membership_id;
                      branchInformation.changed_user_id = User.Identity.GetUserId();
                      branchInformation.changed_date = DateTime.Now;
                      branchFactory.Add(branchInformation);
                      branchFactory.Save();

                      return Json(new { success = true, data = "Successfully Submited" }, JsonRequestBehavior.AllowGet);
                  }
                  else
                  {
                      return Json(new { success = false, errorMessage = "Branch already exists!!!!" }, JsonRequestBehavior.AllowGet);
                  }
              }
              catch (Exception ex)
              {
                  return Json(new { success = false, errorMessage = ex.Message }, JsonRequestBehavior.AllowGet);
              }
          }      
    }
}